.github/workflows/ci.yaml

```diff
@@ -6,28 +6,57 @@ env:
 on:
   push:
     branches:
-    - main
+      - main
   pull_request:
+  workflow_dispatch:
 
 jobs:
-  build:
+  test:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Setup Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: '1.17'
+      - uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.17"
 
-    - name: Run tests
-      run: go test .
-    - name: Vet
-      run: go vet .
-    - name: Build
-      run: |
-        GOOS=linux GOARCH=arm64 go build -o $BINARY_NAME-arm64 main.go
-        GOOS=linux GOARCH=amd64 go build -o $BINARY_NAME-amd64 main.go
+      - name: Run tests
+        run: go test .
+      - name: Vet
+        run: go vet .
+  upload:
```

Comment:
IIUC structuring this as multiple jobs means that `test` & `upload` will [run in parallel](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#overview). This means that we will trigger a deployment _even if_ the tests are failing, which is probably not what we want!

I'd suggest restructuring this to be a single job, or using `jobs.<job_id>.needs` to avoid this.

Reply:
The `needs` section prevents this and means that upload only runs if `test` passes.

Reply:
Sorry - I blame Monday morning! 🙃